website/docs/reference/widgets/table/inline-editing.md

```diff
@@ -1,279 +1,316 @@
+These modes allow you to choose how to handle the flow of updating, saving, and discarding new values within editable tables.
 
-For example, add a regular expression for entering a name. The name can contain only alphabets and spaces between the first and last name.
+##### Single row
 
-```
-/^[a-z -]+$/i
-```
+Cells can be edited within a single row at a time, and then saved using the Save/Discard column buttons. While a row has pending edits, the user isn't allowed to begin editing other rows. Be sure to configure the Save/Discard buttons' **onSave** event to run a query that updates the table's datasource.
 
-If you enter a value other than an alphabet or space (number of special characters), the widget shows an error message "invalid input."
+When one or more columns of a table are **Editable** in Single row mode, a new table column is injected called `Save/Discard`. This new column contains a Save button and a Discard button, which execute the table's **onSave** and **onDiscard** events when clicked. Style options for these buttons are available in the `Save/Discard` column's settings in the Table widget's properties pane.
 
-Similarly, you can use `s.` to only display words that start with the letter "**s**".
+![](</img/SaveDiscard.jpg>)
 
-#### Valid
+The `Save/Discard` column can't be deleted, only hidden; if the update mode is switched to Multi row or if editing is turned off in the table, this column automatically disappears.
 
-Valid property shows the validity of the cell. When the expression evaluates to `false`, the input is considered invalid and the widget shows an Error Message. The following variables are available for binding:
+:::note
+While you are configuring the **onSave** or **onDiscard** events, you can use the `updatedRow` property to access the updated row data.
+:::
 
-1. `currentRow` - gives access to the row values of the editable cell.
-2. `currentIndex` - index of the current editable row.
-3. `editedValue` - the newly entered value on the editable input.
+As an alternative to using the `Save/Discard` buttons and events, you can configure the **onSubmit** event in each column's settings to run a query that saves the new data. The onSubmit event takes place whenever the user clicks away from the edited cell, or presses the Enter key within it.
 
-For example, imagine you want the updated value to be `John`. In the **Valid** property field, add:
+After configuring your save button to send data back to your datasource, set up a success callback nested within the button's onClick to automatically pull the latest information into the table. In this example, `myAPI_get` is a query that refreshes the table data.
 
+```javascript
+{{
+  // in the button's onClick field
+  myAPI_update.run(
+    () => myAPI_get.run(), // success callback
+    () => {} // error callback
+  )
+}}
+
+###### updatedRow
```

Comment:
@spark-c this is not being rendered accurately in the dp. Can you check once?

Reply:
Oh heck, great catch. I was missing the backticks to terminate the code block -- Fixed